elfpy/utils/logs.py

```diff
@@ -116,18 +114,97 @@ def _prepare_log_path(log_filename: str):
     return log_dir, log_name
 
 
-def _create_file_handler(log_dir: str, log_name: str, log_formatter: logging.Formatter, max_bytes: int):
-    """Create a file handler for the given log file"""
+def _create_formatter(log_format_string: str | None = None):
+    """Create Formatter object from a log format string, applying default settings if log_format_string is None."""
+    if log_format_string is None:
+        log_formatter = logging.Formatter(elfpy.DEFAULT_LOG_FORMATTER, elfpy.DEFAULT_LOG_DATETIME)
+    else:
+        log_formatter = logging.Formatter(log_format_string, elfpy.DEFAULT_LOG_DATETIME)
+    return log_formatter
+
+
+def _create_log_level(log_level):
+    """Create log level, applying default settings if log_level is None."""
+    if log_level is None:
+        log_level = elfpy.DEFAULT_LOG_LEVEL
+    return log_level
+
+
+def get_root_logger(root_logger: logging.Logger | None = None) -> logging.Logger:
+    """Retrieve root logger used for elf-simulations, isolated from other loggers (like pytest)."""
+    if root_logger is None:
+        root_logger = logging.getLogger()
+    return root_logger
+
+
+def add_stdout_handler(
+    logger: logging.Logger | None = None,
+    log_format_string: str | None = None,
+    log_level: int | None = logging.INFO,
+    keep_previous_handlers: bool = True,
+):
+    """Add a stdout handler to the root logger."""
+    logger = get_root_logger(logger)
+    if not keep_previous_handlers:
+        _remove_handlers(logger)
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setLevel(_create_log_level(log_level))
+    stream_handler.setFormatter(_create_formatter(log_format_string))
+    logger.addHandler(stream_handler)
+
+
+def add_file_handler(
+    log_filename: str,
+    logger: logging.Logger | None = None,
+    delete_previous_logs: bool = False,
+    log_format_string: str | None = None,
+    log_level: int | None = logging.INFO,
+    max_bytes=None,
+    keep_previous_handlers: bool = True,
+):
+    """Add a file handler to the root logger."""
+    # pylint: disable=too-many-arguments
+    logger = get_root_logger(logger)
+    if not keep_previous_handlers:
+        _remove_handlers(logger)
+    if max_bytes is None:
+        max_bytes = elfpy.DEFAULT_LOG_MAXBYTES
+    log_dir, log_name = _prepare_log_path(log_filename)
+    # Delete the log file if requested
+    if delete_previous_logs and os.path.exists(os.path.join(log_dir, log_name)):
+        os.remove(os.path.join(log_dir, log_name))
+    file_handler = _create_file_handler(
+        log_dir, log_name, _create_formatter(log_format_string), max_bytes, _create_log_level(log_level)
+    )
+    logger.addHandler(file_handler)
+
+
+def _remove_handlers(logger: logging.Logger):
```

Comment:
might as well keep all of these public until we have a good reason to make them private